core/domain/suggestion_services.py

```diff
@@ -1763,15 +1772,15 @@ def update_translation_suggestion(
         suggestion_id: str. The id of the suggestion to be updated.
         translation_html: str. The new translation_html string.
     """
-    suggestion = get_suggestion_by_id(suggestion_id)
+    # Here we use cast because we are narrowing down the type from
+    # BaseSuggestion to SuggestionTranslateContent, because this function
+    # is called only for translate_content suggestions.
+    suggestion = cast(
```

Comment:
Could this be replaced by something like: retrieve the suggestion; check its type; raise error if it's the wrong type?

Reply:
Done